File: sort-colors.ts
{
    function sortColors(nums: number[]): void {
        const dp = new Array(3).fill(0);

        for (const num of nums) {
            dp[num] += 1;
        }

        let start = 0;

        for (let i = 0; i < dp.length; i++) {
            const new_array = Array.from({length: dp[i]}, () => i);

            nums.splice(start, dp[i], ...new_array);
            start += dp[i];
        }

    };

    console.log(sortColors([2,0,1]))
}

File: sort-colors.rs
pub fn sort_colors(nums: &mut Vec<i32>) {
    let mut dp = vec![0; 3];

    for &num in nums.iter() {
        dp[num as usize] += 1;
    }

    let mut index = 0;

    for color in 0..3 {
        for _ in 0..dp[color] {
            nums[index] = color as i32;
            index += 1;
        }
    }
}
